spring-integration-file/src/main/java/org/springframework/integration/file/HeadDirectoryScanner.java

```diff
@@ -39,8 +39,8 @@
 
 	public HeadDirectoryScanner(int maxNumberOfFiles) {
 		HeadFilter headFilter = new HeadFilter(maxNumberOfFiles);
-		this.setFilter(headFilter);
 		this.headFilter = headFilter;
+		this.setFilter(headFilter);
 	}
```

Comment:
Some tests just fail for me with `NPE`. 
Not sure how that has worked before, but here is my analysis:
1. `HeadDirectoryScanner` ctor invokes `this.setFilter(headFilter);`
2. If you take a look properly to the `setFilter()` impl, you'll see this code:
```
compositeFilter.addFilter(filter).addFilter(this.headFilter);
```
where `this.headFilter` is `null` yet.
3. According to the travel to the "deep" by code, we find this code in the `CompositeFileListFilter`:
```
this.fileFilters.addAll(filtersToAdd);
```
where `filtersToAdd` is `null`, of course. So, we just place a `null` to the `Set` alongside with that `headFilter`.

Swapping those operators in the ctor helps me to populate to the `this.fileFilters` above the same object which is collapsed by the `Set` nature.